src/attr/_make.py

```diff
@@ -1924,6 +1925,45 @@ def _assign_with_converter(attr_name, value_var, has_on_setattr):
     )
 
 
+if PY2:
+
+    def _unpack_kw_only_py2(attr_name, default=None):
+        """
+        Unpack *attr_name* from _kw_only dict.
+        """
+        if default is not None:
+            arg_default = ", %s" % default
+        else:
+            arg_default = ""
+        return "%s = _kw_only.pop('%s'%s)" % (
+            attr_name,
+            attr_name,
+            arg_default,
+        )
+
+    def _unpack_kw_only_lines_py2(kw_only_args):
+        """
+        Unpack all *kw_only_args* from _kw_only dict and handle errors.
```

Comment:
Could you please elaborate/outline how it works? Because it's not entirely clear to me and I don't want to figure it out whenever I look at the code. :)

(leave the first like. Just add a blank line and explain what you're doing.)

Reply:
Done, hope it's not too confusing.